File: sentinel-backend/tests/archived-results/analyzer1.js
const fs = require('fs');
const path = require('path');
const axios = require('axios');

function getApiKey() {
    return process.env.API_KEY;
}

async function fetchData(url) {
    try {
        const response = await axios.get(url);
        if (response.status === 200) {
            return response.data;
        } else {
            console.error(`Failed to fetch data: ${response.status}`);
            process.exit(1);
        }
    } catch (error) {
        console.error(`Error fetching data from ${url}:`, error);
        process.exit(1);
    }
}

function parseJson(data) {
    try {
        return JSON.parse(data);
    } catch (e) {
        console.error('Failed to parse JSON:', e.message);
        process.exit(1);
    }
}

function validateUrl(url) {
    const parsed = new URL(url);
    if (!parsed.protocol || !parsed.hostname) {
        throw new Error("Invalid URL");
    }
}

function logActivity(message, level = 'info') {
    const logFile = '/var/log/app.log';
    const timestamp = new Date().toISOString();
    const entry = `[${timestamp}][${level}] ${message}\n`;
    fs.appendFileSync(logFile, entry);
}

async function main() {
    const apiKey = getApiKey();
    if (!apiKey) {
        console.error('API key is required.');
        process.exit(1);
    }

    const url = prompt('Enter the URL to fetch data from: ');
    validateUrl(url);  // Vulnerability: No error handling after validation

    try {
        const rawData = await fetchData(url);
        const parsedData = parseJson(rawData);

        for (const item of parsedData.items) {
            if (!item.id || !parseInt(item.id)) continue;

            // Potential SQL Injection
            require('sqlite3').verbose().open(item.id);  // Vulnerability: Direct use of input as database name

            // Potential Path Traversal
            const filePath = path.join('/data/', item.path);
            fs.readFileSync(filePath, 'utf8');  // Vulnerability: Lack of path validation

            logActivity(`Processed item ${item.id} from ${url}`);
        }
    } catch (e) {
        console.error('An error occurred:', e.message);
        process.exit(1);
    }
}

main();

File: sentinel-backend/tests/archived-results/analyzer1.py
import os
import sys
import json
import requests
from datetime import datetime
from urllib.parse import urlparse

def get_api_key():
    return os.environ.get('API_KEY', '')

def fetch_data(url):
    try:
        response = requests.get(url)
        if response.status_code == 200:
            return response.text
        else:
            print(f"Failed to fetch data: {response.status_code}")
            sys.exit(1)
    except Exception as e:
        print(f"Error fetching data from {url}: {e}")
        sys.exit(1)

def parse_json(data):
    try:
        return json.loads(data)
    except ValueError as e:
        print(f"Failed to parse JSON: {e}")
        sys.exit(1)

def validate_url(url):
    parsed = urlparse(url)
    if not all([parsed.scheme, parsed.netloc]):
        raise ValueError("Invalid URL")

def log_activity(message, level='info'):
    log_file = '/var/log/app.log'
    with open(log_file, 'a') as f:
        timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
        log_entry = f"[{timestamp}][{level}] {message}\n"
        f.write(log_entry)

def main():
    api_key = get_api_key()
    if not api_key:
        print("API key is required.")
        sys.exit(1)
    
    url = input("Enter the URL to fetch data from: ")
    validate_url(url)  
    
    try:
        raw_data = fetch_data(url)
        parsed_data = parse_json(raw_data)

        for item in parsed_data['items']:
            item_id = item.get('id', '')
            if not item_id.isdigit():
                continue

           
            conn = sqlite3.connect(item_id)  
            
            
            with open(os.path.join('/data/', item['path']), 'r') as f:
                file_content = f.read()  

            log_activity(f"Processed item {item_id} from {url}")
            
    except Exception as e:
        print(f"An error occurred: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()
